app/utils/kci_test/__init__.py

```diff
@@ -446,82 +433,80 @@ def _update_test_group_doc_ids(group_doc, database):
     build_doc = utils.db.find_one2(database[models.BUILD_COLLECTION], spec)
 
     if job_doc:
-        group_doc.job_id = job_doc.get(models.ID_KEY, None)
+        group_doc.job_id = job_doc.get(models.ID_KEY)
     else:
         utils.LOG.warn(
             "No job document found for test group %s-%s-%s-%s (%s)",
             job, branch, kernel, defconfig_full, arch)
 
     if build_doc:
         doc_get = build_doc.get
-        group_doc.build_id = doc_get(models.ID_KEY, None)
+        group_doc.build_id = doc_get(models.ID_KEY)
 
         # In case we do not have the job_id key with the previous search.
-        if all([not group_doc.job_id, doc_get(models.JOB_ID_KEY, None)]):
-            group_doc.job_id = doc_get(models.JOB_ID_KEY, None)
+        if all([not group_doc.job_id, doc_get(models.JOB_ID_KEY)]):
```

Comment:
Wouldn't that be easier to read and understand with a simple condition like 
`if not group_doc.job_id and doc_get(models.JOB_ID_KEY, None):`?

Reply:
Yes, that seems better.  I didn't try to simplify the code other than remove unnecessary `None` default values and unused fields in the test group model.

Reply:
@mgalka I've actually simplified this quite a bit, see new version.

Reply:
This is even better :) Thanks